docs/how-it-works/internals.md

```diff
@@ -0,0 +1,120 @@
+
+In my opinion, this presents performance pitfalls and gives Pydeck a stifling amount of flexibility. The Python API isn't able to do much of anything because the user input could be _anything_ and users end up writing a sort of JavaScript in Python strings.
+
+## Lonboard
+
+### Goals and target user persona
+
+While I was never a primary pydeck developer and can't speak to their goals, I can say that Lonboard has a well-formed idea of the target user.
+
+When I was learning geospatial data processing in Python, I learned that GeoPandas, Shapely, and others were amazing tools. But I would perform operations and not have a _deep understanding_ of what the output looked like. I wanted some way to quickly get a picture of what my data looked like. This drove me to build [`keplergl_cli`](https://github.com/kylebarron/keplergl_cli), a single function or CLI command to visualize data in an interactive HTML map. The desire for this simplicity lives on in Lonboard's `viz` function.
+
+This experience informs the target user of Lonboard:
+
+- A Python user who does not know or want to know JavaScript.
+- Wants fast exploratory data analysis.
+- Already working with data in Python.
+- A reasonably fast internet connection on a desktop computer.
+
+### Python classes
+
+The `Map` class is the primary and only widget associated with JavaScript that renders anything. `Map` subclasses from `anywidget.AnyWidget`, which dynamically fetches Lonboard's JavaScript ESM bundle. The `Map` class emulates deck.gl's `Deck` class. It synchronizes map state and can be passed a sequence of `Layer` objects.
+
+`Layer` classes are designed to map to each underlying deck.gl layer. Most `Layer` classes subclass from `BaseArrowLayer`, which stores a `pyarrow.Table` in the dataclass. This data object gets serialized to Parquet and rendered in a layer provided by `@geoarrow/deck.gl-layers` as described later on. The primary API for each of these Arrow-based layers is `from_geopandas`, which converts a `GeoDataFrame` to a `pyarrow.Table` with a [GeoArrow](https://geoarrow.org/)-represented geometry column. A few layers are not Arrow-based and do not subclass from `BaseArrowLayer`: primarily `BitmapLayer` and `BitmapTileLayer`. These pass user data input directly into a core deck.gl layer.
+
+These `Layer` classes subclass from `ipywidgets.Widget` but are _not_ associated with any JavaScript of their own. But by being widgets themselves, their data is synced with JavaScript and Python data changes are propagated as events to the core `Map` object.
+
+Each layer has parameters that map to deck.gl accessors and render properties. Each of these has validation on the Python side to catch invalid input early. Accessors are modeled with ["trait" objects](https://developmentseed.org/lonboard/latest/api/traits/). Scalar accessors are stored as a scalar int/string/float. Function accessors are evaluated in Python and stored as a `pyarrow.Array`.
```

Comment:
> Function accessors are evaluated in Python

This is analogous to [supplying attributes directly](https://deck.gl/docs/developer-guide/performance#supply-attributes-directly) in JavaScript, correct? In that sense they are not so much "function accessors" but rather pre-computed attributes. IMHO the name "accessor" doesn't work well here, if functions are not supported

Reply:
> > Function accessors are evaluated in Python
> 
> This is analogous to [supplying attributes directly](https://deck.gl/docs/developer-guide/performance#supply-attributes-directly) in JavaScript, correct? In that sense they are not so much "function accessors" but rather pre-computed attributes. IMHO the name "accessor" doesn't work well here, if functions are not supported

I'm using the name "accessor" to provide consistency with the meaning of the JS API. In this context I'm using "accessor" to refer to any property that starts with the `get` prefix.

I've considered whether to emulate the functional API on the Python side. Where the user could pass 

```py
ScatterplotLayer(
    get_radius=lambda object: object["value"] * 1000
)
```

but I'm hesistant to do this because this requires a slow Python for loop. If we nudge the user to do 

```py
ScatterplotLayer(
    get_radius=dataframe["value"] * 1000
)
```
then numpy evaluates that in C, we don't have to rebuild the buffers, and it'll be a lot faster without the user needing to understand the difference between those two.